### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by nes123** [wix/react-native-calendars#1915]
The calendar is doing many re-renderings due to the function definitions in index.js. The Day component is getting re-rendered since the functions onPress and onLongPress are declared as arrow functions.

Here are the lines of code:

```
const onPressDay = useCallback((date) => {
        if (date)
            handleDayInteraction(date, onDayPress);
    }, [handleDayInteraction, onDayPress]);

const onLongPressDay = useCallback((date) => {
        if (date)
            handleDayInteraction(date, onDayLongPress);
    }, [handleDayInteraction, onDayLongPress]);

<Day {...dayProps} date={toMarkingFormat(day)} state={getState(day, currentMonth, props)} marking={markedDates?.[toMarkingFormat(day)]} onPress={onPressDay} onLongPress={onLongPressDay}/>
```

It causes the calendar to open very slowly. Is there a way to overcome this rendering issue?

**Comment by AleksandrTermenzhy** [comment]
Hey!

It looks like the issue with callbacks is fixed with #1891.
But there is still unnecessary re-renders are happening because of `marking` prop. `<Day />` component itself is wrapped with `memo` but as long as it receive complex props we have to pass comparison function as the second argument to `memo`. At least I've managed to avoid those re-renders by passing `isEqual` from `lodash`.

There is might be a better way to optimize it.

@Inbal-Tish could you please take a look?

**Comment by zhGio** [comment]
Definitely not considering this stale..
